Library/Homebrew/cmd/shellenv.sh

```diff
@@ -32,31 +32,67 @@ homebrew-shellenv() {
       echo "set -gx HOMEBREW_PREFIX \"${HOMEBREW_PREFIX}\";"
       echo "set -gx HOMEBREW_CELLAR \"${HOMEBREW_CELLAR}\";"
       echo "set -gx HOMEBREW_REPOSITORY \"${HOMEBREW_REPOSITORY}\";"
-      echo "set -q PATH; or set PATH ''; set -gx PATH \"${HOMEBREW_PREFIX}/bin\" \"${HOMEBREW_PREFIX}/sbin\" \$PATH;"
+      if [[ ":${HOMEBREW_PATH}:" != *":${HOMEBREW_PREFIX}/bin:"* ]] && [[ ":${HOMEBREW_PATH}:" != *":${HOMEBREW_PREFIX}/sbin:"* ]]
```

Comment:
The original `if [[ "${HOMEBREW_PATH%%:"${HOMEBREW_PREFIX}"/sbin*}" == "${HOMEBREW_PREFIX}/bin" ]]` is intended to do this. No need to duplicate it for each `case` statement, instead amend that instead. I'd also be interested to know why that isn't working for you.

Reply:
I thought that might be the case. I'd like to export "$HOMEBREW_PREFIX" specifically because we have people on both Intel and Apple Silicon macs and I'm trying to write scripts for them that run on both and I've been using the `shellenv` command to set that up instead of maintaining my own logic to do it.

Maybe a better solution is to check if /bin and /sbin are in the path (maybe updating the conditional to make it order-agnostic?) and if they are, echo the non-mutative ENV vars and return?

Reply:
Could consider combining the checks and setting their status into local variables e.g. `HOMEBREW_PATH_HAS_BIN`,`HOMEBREW_PATH_HAS_SBIN`, etc. to avoid the repetition.

The important things here are:
- avoiding repetition
- ensuring `eval $(brew shellenv); eval $(brew shellenv)` will be a faster/no-op the second time